providers/r2dbc/shedlock-provider-r2dbc-internal/src/main/java/net/javacrumbs/shedlock/provider/r2dbc/internal/AbstractR2dbcStorageAccessor.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2009 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.javacrumbs.shedlock.provider.r2dbc.internal;
+
+import io.r2dbc.spi.R2dbcDataIntegrityViolationException;
+import io.r2dbc.spi.Statement;
+import net.javacrumbs.shedlock.core.ClockProvider;
+import net.javacrumbs.shedlock.core.LockConfiguration;
+import net.javacrumbs.shedlock.support.AbstractStorageAccessor;
+import net.javacrumbs.shedlock.support.LockException;
+import net.javacrumbs.shedlock.support.annotation.NonNull;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+
+import java.time.Instant;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Internal class, please do not use.
+ */
+public abstract class AbstractR2dbcStorageAccessor extends AbstractStorageAccessor {
```

Comment:
What's thereason for extracting it to a new module? I would prefer to keep it package-protected inside the r2dbc module until it's needed elsewhere.

Reply:
I did it for the same reason as jdbc did - to provide the availability to implement lock provider via spring-r2dbc wrapper in another module in future.